### PR TITLE
dong feed back - #856

### PR DESCRIPTION
**Pull request by phongvu2812**
# Ticket
#856

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

**Review by vietdangquoc — APPROVED** [review]

File: dong-service/handlers/user_payment_handler.go [hunk 1]
```diff
@@ -37,6 +38,19 @@ func (h *UserPaymentHandler) UpdatePaymentInfo(c *gin.Context) {
 		return
 	}
 
+	if req.AccountNumber == "" {
```
Comment (hoangdohuy) [inline dong-service/handlers/user_payment_handler.go:54]: Có nên validate độ dài cho đỡ bị spam linh tinh không?

File: dong-service/handlers/user_payment_handler.go [hunk 2]
```diff
@@ -37,6 +38,19 @@ func (h *UserPaymentHandler) UpdatePaymentInfo(c *gin.Context) {
...
+		c.JSON(http.StatusBadRequest, models.ErrorResponse(http.StatusBadRequest, "Please enter the account number"))
+		return
+	}
+	if !regexp.MustCompile(`^\d+$`).MatchString(req.AccountNumber) {
```
Comment (hoangdohuy) [inline dong-service/handlers/user_payment_handler.go:62]: Có validate cả bankname không? vì mình chỉ hỗ trợ 1 vài bank?

Comment (QTR33) [reply dong-service/handlers/user_payment_handler.go:62]: done

File: dong-service/handlers/user_payment_handler.go (hunk 1 above)
Comment (QTR33) [reply dong-service/handlers/user_payment_handler.go:54]: đã giới hạn 14 ký tự

File: dong-service/handlers/user_payment_handler.go [hunk 3]
```diff
@@ -37,6 +38,35 @@ func (h *UserPaymentHandler) UpdatePaymentInfo(c *gin.Context) {
...
+	if !validBanks[req.BankName] {
+		c.JSON(http.StatusBadRequest, models.ErrorResponse(http.StatusBadRequest, "Invalid bank name"))
+		return
+	}
+
+	if req.AccountNumber == "" {
+		c.JSON(http.StatusBadRequest, models.ErrorResponse(http.StatusBadRequest, "Please enter the account number"))
+		return
+	}
+	if len(req.AccountNumber) > 14 {
```
Comment (hoangdohuy) [inline dong-service/handlers/user_payment_handler.go]: hardcode

File: dong-service/handlers/user_payment_handler.go [hunk 4]
```diff
@@ -37,6 +38,35 @@ func (h *UserPaymentHandler) UpdatePaymentInfo(c *gin.Context) {
...
+		c.JSON(http.StatusBadRequest, models.ErrorResponse(http.StatusBadRequest, "Account number must be 14 digits or less"))
+		return
+	}
+	if !regexp.MustCompile(`^\d+$`).MatchString(req.AccountNumber) {
+		c.JSON(http.StatusBadRequest, models.ErrorResponse(http.StatusBadRequest, "Account number must contain only digits"))
+		return
+	}
+	if req.AccountName == "" {
```
Comment (hoangdohuy) [inline dong-service/handlers/user_payment_handler.go:66]: cũng tương tự như account number cũng nên giới hạn lại chỉ cho 50 100 ký tự gì đó thôi

File: frontend/modules/p2p/components/p2p-trading/create-offer-form/trade-type-section.tsx [hunk 5]
```diff
@@ -82,7 +47,7 @@ export const TradeTypeSection = ({ control, trigger }: TradeTypeSectionProps) =>
                   <>
                     <Input
                       type="text"
-                      placeholder="1"
+                      placeholder="0"
```
Comment (hoangdohuy) [inline frontend/modules/p2p/components/p2p-trading/create-offer-form/trade-type-section.tsx]: update place holder -> value

**Review by hoangdohuy — APPROVED** [review]
LGTM
